docs/start/framework/route-module.md

```diff
@@ -39,6 +39,32 @@ export default function MyRouteComponent() {
   );
 }
 ```
+### Default props passed to the Component
+1. params: An object containing the route parameters (if any).
+
+2. location: An object providing details about the current location (e.g., pathname, search, etc.).
+
+3. navigate: A function to programmatically navigate to another route.
+
+4. loaderData: The data returned from the loader function assigned to the route. eg. A loader might fetch some data from an API and return it for the route to use.
+
+### Using props
+```tsx filename=app/routes/my-route-with-default-params.tsx
+export default function MyRouteComponent({ loaderData, actionData, params, matches }) {
+  return (
+    <div>
+      <h1>Welcome to My Route with Props!</h1>
+      <p>Loader Data: {JSON.stringify(loaderData)}</p>
+      <p>Action Data: {JSON.stringify(actionData)}</p>
+      <p>Route Parameters: {JSON.stringify(params)}</p>
+      <p>Matched Routes: {JSON.stringify(matches)}</p>
+    </div>
+  );
+}
+```
+
+### Benefits of using props
+1. Route data is directly accessible via props, leading to more readable code.
```

Comment:
Another benefit is Route data can be automatically typed if you follow the instructions and import the `Route` from the `+types`